src/main/java/com/iota/iri/service/transactionpruning/jobs/AbstractTransactionPrunerJob.java

```diff
@@ -0,0 +1,131 @@
+package com.iota.iri.service.transactionpruning.jobs;
+
+import com.iota.iri.controllers.TipsViewModel;
+import com.iota.iri.service.snapshot.Snapshot;
+import com.iota.iri.service.transactionpruning.TransactionPruner;
+import com.iota.iri.service.transactionpruning.TransactionPrunerJob;
+import com.iota.iri.service.transactionpruning.TransactionPrunerJobStatus;
+import com.iota.iri.service.transactionpruning.TransactionPruningException;
+import com.iota.iri.storage.Tangle;
+
+/**
+ * Implements the most basic functionality that is shared by the different kinds of jobs.
+ */
+public abstract class AbstractTransactionPrunerJob implements TransactionPrunerJob {
+    /**
+     * Holds the execution status of the job.
+     */
+    private TransactionPrunerJobStatus status = TransactionPrunerJobStatus.PENDING;
+
+    /**
+     * Holds a reference to the manager of the job that schedules it execution.
+     */
+    private TransactionPruner transactionPruner;
+
+    /**
+     * Holds a reference to the tangle object which acts as a database interface.
+     */
+    private Tangle tangle;
+
+    /**
+     * Holds a reference to the manager for the tips (required for removing pruned transactions from this manager).
+     */
+    private TipsViewModel tipsViewModel;
+
+    /**
+     * Holds a reference to the last local or global snapshot that acts as a starting point for the state of ledger.
+     */
+    private Snapshot snapshot;
+
+    /**
+     * {@inheritDoc}
+     */
```

Comment:
Since i use sublime sometimes which kind of needs it i started adding it everywhere where it makes sense - imho better be specific than to to rely on a certain behaviour of the IDE. Having a javadoc section everywhere also allows me to add implementation specific comments, which would otherwise overwrite the default docs.